### PR TITLE
Xray metrics backend and API endpoint

### PR DESCRIPTION
**Pull request by sbelak**
Adds Xrays for metrics.

### Todo
[ ] ~~UI~~
[ x ] tests (see below) 

### Discussion

* How aggressive should we be with dimension removal? Currently we limit ourselves to dimensions that are either numbers, date times, or categories; and are not named ID. My feeling is we could be more aggressive with exclusions of specific type signatures.
* What should the default date time resolution be. Currently it's set to date. Should it be changeable? We could alo feature extraction at multiple resolutions and pick the most interesting one (at least for default).

~~Note: the awkward casting to Card will disappear once GH-6035 gets merged. Ditto for tests.~~

### CODE REVIEW

**Comment by sbelak** [comment]
Minimized this to just the backend part so we don't have such a big backlog of open PRs.

**Comment by sbelak** [comment]
I'm still not 100% on reusing Query model to hold the ad hoc queries. Would it be better to have a wrapper model? Should relationship between Card and Query be encoded in a (type) hierarchy?

File: src/metabase/feature_extraction/core.clj [hunk 1]
```diff
@@ -115,6 +118,46 @@
...
+                         [(:name field)
+                          (->> field
+                               field->breakout
+                               vector
+                               (assoc-in query [:dataset_query :query :breakout])
+                               (extract-features opts))]))]
+    {:constituents constituents
+     :features     {:metric metric
+                    :table  (Table table_id)}
+     :samlple?     (some (comp :sample? val) constituents)}))
```
Comment (salsakran) [inline src/metabase/feature_extraction/core.clj]: sample is mispelled.

**Review by salsakran — APPROVED** [review]
seems reasonable, aside from a mispelling.

**Pull request merged**